### ISSUE TITLE
[slp-vectorizer] Assertion `Mask[I] == UndefMaskElem && "Multiple uses of scalars."' failed

### ISSUE THREAD

**Issue by vedantparanjape-amd** [llvm/llvm-project#61018]
While running a benchmark using LLVM-16 upstream, I encountered an assertion. Attached the log here and a reproducable code snippet.

```bash
clang-16: /home/amd/llvm-project/llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp:8147: T* performExtractsShuffleAction(llvm::MutableArrayRef<std::pair<T*, llvm::SmallVector<int> > >, llvm::Value*, llvm::function_ref<unsigned int(T*)>, llvm::function_ref<std::pair<T*, bool>(T*, llvm::ArrayRef<int>, bool)>, llvm::function_ref<T*(llvm::ArrayRef<int>, llvm::ArrayRef<U*>)>) [with T = const llvm::slpvectorizer::BoUpSLP::TreeEntry]: Assertion `Mask[I] == UndefMaskElem && "Multiple uses of scalars."' failed.
PLEASE submit a bug report to the technical support section of https://developer.amd.com/amd-aocc and include the crash backtrace, preprocessed source, and associated run script.
```
To run the same, please you the following command

```bash
opt -passes='slp-vectorizer' fail5.2.ll -o tmp.ll
```
Godbolt link with the reduced test case: https://llvm.godbolt.org/z/rW6x14jc1

Earlier there was a ticket which fixed the same assertion, [patch](https://reviews.llvm.org/rG85f6b15ee50feb316047f52d4bd6ddc639e3c5c1#change-6eMSUXjvvi9P) and [ticket](https://reviews.llvm.org/D115750)

CC @alexey-bataev

**Comment by vedantparanjape-amd** [comment]
@llvmbot

**Comment by alexbatashev** [comment]
@vedantparanjape-amd you're looking for @alexey-bataev

**Comment by vedantparanjape-amd** [comment]
> @vedantparanjape-amd you're looking for @alexey-bataev

Oops, sorry for the ping.

**Comment by alexey-bataev** [comment]
Fixed in 007177bdde60a822ccaefc96593e818c2235e2d5

**Comment by vedantparanjape-amd** [comment]
Verified locally, this fixed the problem. Thanks @alexey-bataev
